src/metabase/db/setup.clj

```diff
@@ -120,6 +124,36 @@
       (log/infof "Successfully verified %s %s application database connection. %s"
                  (.getDatabaseProductName metadata) (.getDatabaseProductVersion metadata) (u/emoji "✅")))))
 
+(mu/defn- check-encryption
+  "Ensure encryption env variable is correctly set if needed, and encrypt the database if it needs to be
+  Encryption status is tracked by an 'encryption-check' value in the settings table.
+  NOTE: the encryption-check setting is not managed like most settings with 'defsetting' so we can manage checking the raw values in the database"
+  [db-type :- :keyword
+   data-source :- (ms/InstanceOfClass javax.sql.DataSource)]
+  (when (or
+         (liquibase/table-exists? "setting" (.getConnection ^javax.sql.DataSource data-source))
+         (liquibase/table-exists? "SETTING" (.getConnection ^javax.sql.DataSource data-source)))
+    (try
+      (t2/insert! :conn data-source :setting {:key "encryption-check" :value "unencrypted"})
```

Comment:
You should be able to leave out the `:conn` argument here since toucan2 will use `*application-db*` (bound on L196) as the default connection
```suggestion
      (t2/insert! :setting {:key "encryption-check" :value "unencrypted"})
```